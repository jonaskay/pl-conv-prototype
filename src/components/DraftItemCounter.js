import React from 'react';

class DraftItemCounter extends React.Component {
  render() {
    const itemCount = this.props.items.length;

    return (
      <span>{itemCount} {itemCount === 1 ? 'track' : 'tracks'}</span>
    );
  }
}

export default DraftItemCounter;
