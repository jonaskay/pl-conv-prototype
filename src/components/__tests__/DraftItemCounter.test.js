import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import DraftItemCounter from '../DraftItemCounter';

it('renders item count for 0 items', () => {
  const component = ReactTestRenderer.create(<DraftItemCounter items={[]} />);
  expect(component).toMatchSnapshot();
})

it('renders item count for 1 item', () => {
  const component = ReactTestRenderer.create(<DraftItemCounter items={['foo']} />);
  expect(component).toMatchSnapshot();
})

it('renders item count for multiple items', () => {
  const component = ReactTestRenderer.create(<DraftItemCounter items={['foo', 'bar']} />);
  expect(component).toMatchSnapshot();
})
