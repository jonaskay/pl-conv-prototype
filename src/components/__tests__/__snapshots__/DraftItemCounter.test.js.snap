// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item count for 0 items 1`] = `
<span>
  0
   
  tracks
</span>
`;

exports[`renders item count for 1 item 1`] = `
<span>
  1
   
  track
</span>
`;

exports[`renders item count for multiple items 1`] = `
<span>
  2
   
  tracks
</span>
`;
